src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -69,18 +72,25 @@
             // BEGIN string is 11 chars long
             throw log.malformedPemContent(matcher.start() + 11);
         }
-        if (! pemContent.contentEquals("-----")) {
+        if (! pemContent.limitedTo(5).contentEquals("-----")) {
             throw log.malformedPemContent(pemContent.offset());
         }
-        final ByteIterator byteIterator = pemContent.delimitedBy('-').base64Decode();
+
+        String encodedKey = pemContent.delimitedBy('-').drainToString();
+
+        encodedKey = encodedKey.replaceAll("\r\n", "");
+        encodedKey = encodedKey.replaceAll("\n", "");
+        encodedKey = encodedKey.trim();
+
+        final ByteIterator byteIterator = ByteIterator.ofBytes(encodedKey.getBytes()).base64Decode();
```

Comment:
Hmm I'm not excited about this one.  Maybe we need to do something like the following:

```
final ByteIterator byteIterator = pemContent.skipSpaces().delimitedBy('-').base64Decode();
```

...and add a ``skipSpaces()`` method to CodePointIterator?  If that's too much trouble maybe just change the last line:

```
final ByteIterator byteIterator = CodePointIterator.ofString(encodedKey).base64Decode();
```

to avoid the byte array creation, and before that, use a one-pass regular expression on the key:
```
encodedKey = encodedKey.replaceAll("\\s+", "");
```